test/pool.test.ts

```diff
@@ -298,6 +298,7 @@ test(
     async function user(pool: Pool): Promise<void> {
       const proxy = await pool.acquire();
 
+      // @ts-ignore
```

Comment:
I'd suggest using `@ts-ignore` on as small operations as possible.  E.g. instead of 

```
// @ts-ignore
if (connections.indexOf(proxy.connection) === -1) {
```

do

```
// @ts-ignore
const proxyCon = proxy.connection;

if (connections.indexOf(proxyCon) === -1) {
```


This way you're minimizing the impact of disabling typing.

Reply:
Thanks for the heads up. By the way, what do you think about using @ts-ignore to keep the `connectionHolder`'s proxy private?

```ts
    // @ts-ignore
    if (connectionProxy.holder.pool !== this) {
```

I like it on one hand, but I feel it's a compromise.

Reply:
I had the same thought. Let's do that. Essentially we'd be emulating C++'s friendly classes this way.

Reply:
OTOH we can do differently by using symbols and not resorting to `@ts-ignore`:

```
const getConnection = Symbol.for('connection getter');

class Foo {
  [getConnection] = () => {
  }
}
```

Now because `getConnection` is not exported it can be used privately in our edgedb codebase, but everyone else wouldn't be able to access it.

Reply:
This gives you the benefit of continuing to use strict typing.